app/src/androidTest/java/org/openobservatory/ooniprobe/client/OONIAPIClientTest.java

```diff
@@ -64,6 +72,51 @@ public void onError(String msg) {
     }
 
     @Test
-    public void testDeleteJsons() {
+    public void testSelectMeasurementsWithJson() {
+        Delete.table(Measurement.class);
+        addMeasurement(EXISTING_REPORT_ID, true);
+        addMeasurement(EXISTING_REPORT_ID_2, true);
+        addMeasurement(NONEXISTING_REPORT_ID, true);
+        addMeasurement(NONEXISTING_REPORT_ID, true);
+        List<Measurement> measurements = Measurement.selectMeasurementsWithJson(a);
+        if (measurements.size() == 2 &&
+                containsMeasurement(measurements, EXISTING_REPORT_ID) &&
+                containsMeasurement(measurements, EXISTING_REPORT_ID_2))
+            Assert.assertTrue(true);
+
+        Assert.fail();
+    }
+
+    private Boolean containsMeasurement(List<Measurement> measurements, String report_id){
+        for (int i = 0; i < measurements.size(); i++) {
+            Measurement measurement = measurements.get(i);
+            if (measurement.report_id.equals(report_id))
+                return true;
+        }
+        return false;
     }
+
+    private Measurement addMeasurement(String report_id, Boolean write_file) {
+        //Simulating measurement done and uploaded
+        Measurement measurement = new Measurement();
+        measurement.report_id = report_id;
+        measurement.is_done = true;
+        measurement.is_uploaded = true;
+        measurement.save();
+        if (write_file){
+            File entryFile = Measurement.getEntryFile(c, measurement.id, measurement.test_name);
+            entryFile.getParentFile().mkdirs();
+            try {
+                FileUtils.writeStringToFile(
+                        entryFile,
+                        "",
+                        Charset.forName("UTF-8")
+                );
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
I believe you want the error to propagate in this case, since that's only used for tests and we want to make sure that, if we wanted to write a file during tests, we did really write a file. Isn't it?

Reply:
If we don't write a file the test will fail, simple as that. Do we need to do something more?